tasks/drupal/deploy.yml

```diff
@@ -0,0 +1,164 @@
+---
+
+### Build
+
+- name: Create project folder structure Drupal
+  file:
+    path: "{{ steamengine_drupal_persistent_path }}/{{ item.key }}"
+    state: directory
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: 0770
+  with_dict: "{{ steamengine_drupal_persistent_directory }}"
+
+- name: Download build {{ steamengine_build_url }}
+  get_url:
+    url: "{{ steamengine_build_url }}"
+    dest: "{{ steamengine_project_root_path }}/project.zip"
+    checksum: "{{ steamengine_build_checksum }}"
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+    headers: "{{ steamengine_build_url_headers }}"
+    validate_certs: "{{ steamengine_build_url_validate_certs }}"
+  notify:
+    - "{{ steamengine_project_name }} fixpermissions"
+    - "{{ steamengine_project_name }} restart"
+  register: get_url_build
+  tags:
+    - steamengine_deploy_drupal
+
+- name: "Is there a new build: {{ get_url_build.changed }}"
+  set_fact:
+    new_build_to_deploy: "{{ get_url_build.changed|bool }}"
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Create tmp directory
+  tempfile:
+    state: directory
+    suffix: "{{ steamengine_project_name }}"
+  register: tempdir
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Unarchive build
+  unarchive:
+    src: "{{ steamengine_project_root_path }}/project.zip"
+    dest: "{{ tempdir.path }}"
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_sf
+
+- name: Ensure settings
+  copy:
+    src: "{{ tempdir.path }}/web/sites/default/default.settings.php"
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    force: no
+    remote_src: true
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+    mode: u=rwx,g=rx,o=
+  register: drupal_settings
+  when: new_build_to_deploy
+  tags:
+    - steamengine_deploy_drupal
+
+- name: Customize drupal settings
+  blockinfile:
+    dest: "{{ steamengine_conf_path }}/settings.php"
+    block: "{{ steamengine_drupal_project_setting_block }}"
+    mode: u=rwx,g=rx,o=
+    owner: "{{ steamengine_project_user }}"
+    group: "{{ steamengine_app_user }}"
+  when: new_build_to_deploy
```

Comment:
There is no way to change the config whithout deploying a new version or deleting the project zip inside the project_root directory.

To me this condition is unnecessary and could be remove without breaking indempotence.

Reply:
Still relevant